### PR TITLE
Add documentation for deprecated-class, deprecated-decorator, deprecated-method, deprecated-pragma and deprecated-typing-alias

### PR DESCRIPTION
**Pull request by ollie-iterators**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add documentation for deprecated-class, deprecated-decorator, deprecated-method, deprecated-pragma and deprecated-typing-alias

Refs #5953 

Closes #XXXX

### CODE REVIEW

File: doc/data/messages/d/deprecated-class/pylintrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[main]
+py-version = 3.3
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-class/pylintrc:2]: This one is implicit as we never used py-version before python 3.5 and we're supporting from python3.7 upward.

File: doc/data/messages/d/deprecated-decorator/good.py [hunk 2]
```diff
@@ -1 +1,3 @@
-# This is a placeholder for correct code for this message.
+class MyClass:
+    def my_method(cls):
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-decorator/good.py:2]: We need to use the replacement of ``@abc.abstractclassmethod``

File: doc/data/messages/d/deprecated-pragma/bad.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+class Class:
+    @property
+    @classmethod
+    def func(cls):
+        pass
+
+
+if Class.func:
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-pragma/bad.py:11]:
```suggestion
```

File: doc/data/messages/d/deprecated-pragma/bad.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# pylint: disable-msg=too-few-public-methods,using-constant-test # [deprecated-pragma]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-pragma/bad.py:1]:
```suggestion
# pylint: disable-msg=eval-used # [deprecated-pragma]
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/data/messages/d/deprecated-pragma/bad.py (hunk 4 above)
Comment (ollie-iterators) [reply doc/data/messages/d/deprecated-pragma/bad.py:1]: Changed

File: doc/data/messages/d/deprecated-pragma/good.py [hunk 5]
```diff
@@ -1 +1,9 @@
-# This is a placeholder for correct code for this message.
+class Class:
+    @property
+    @classmethod
+    def func(cls):
+        pass
+
+
+if Class.func:
+    pass
```
[the remark this answers is not in the document]
Comment (ollie-iterators) [reply doc/data/messages/d/deprecated-pragma/good.py:9]: Changed

File: doc/data/messages/d/deprecated-decorator/good.py (hunk 2 above)
Comment (ollie-iterators) [reply doc/data/messages/d/deprecated-decorator/good.py:2]: It is classmethod

File: doc/data/messages/d/deprecated-class/pylintrc (hunk 1 above)
Comment (ollie-iterators) [reply doc/data/messages/d/deprecated-class/pylintrc:2]: The example I found for deprecated-class was said to be for Python 3.3

File: doc/data/messages/d/deprecated-decorator/bad.py [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import abc
+
+
+class FishClass:
+    @abc.abstractclassmethod  # [deprecated-decorator]
+    def my_method(cls):
+        pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-decorator/bad.py:7]:
```suggestion
class Animal:
    @abc.abstractclassmethod  # [deprecated-decorator]
    def breath(cls):
        pass
```

File: doc/data/messages/d/deprecated-decorator/good.py [hunk 7]
```diff
@@ -1 +1,4 @@
-# This is a placeholder for correct code for this message.
+class FishClass:
+    @classmethod
+    def my_method(cls):
+        pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-decorator/good.py:4]:
```suggestion
from abc import abstractmethod, classmethod

class Animal:
    @classmethod
    @abstractmethod
    def breath(cls):
        pass
```

File: doc/data/messages/d/deprecated-pragma/bad.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+class CatClass:
+    @property
+    @classmethod
+    def func(cls):
+        pass
+
+
+if CatClass.func:
+    pass
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-pragma/bad.py:11]:
```suggestion
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/data/messages/d/deprecated-decorator/good.py (hunk 7 above)
Comment (Pierre-Sassoulas) [inline doc/data/messages/d/deprecated-decorator/good.py:4]:
```suggestion
import abc

class Animal:
    @abc.classmethod
    @abc.abstractmethod
    def breath(cls):
        pass
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**